### PR TITLE
Dynamodb table name

### LINKED ISSUE #87: Feature Request - Allow a parameter for the name of DynamoDB table
## Describe the Feature
By default, the name of the DynamoDB Table is `lock`

## Use Case
It will be good to have a customized name just in case I have a different environment and I want to name my table differently. (I know we can use a common table but good to have an option)

## Describe Ideal Solution
A variable like `dynamodb_table_name = "terraform-lock"`

## Alternatives Considered
After running `terraform init`, I went to file `.terraform/modules/terraform_state_backend/main.tf` and added `name       = "terraform"` to this block

```
module "dynamodb_table_label" {
  source     = "cloudposse/label/null"
  version    = "0.22.0"
  attributes = compact(concat(var.attributes, ["lock"]))
  context    = module.this.context
  name       = "terraform"
}
```

### PR DESCRIPTION
**Pull request by nitrocode**
## what
* Add dynamodb table name override

## why
* It's required to have a unique s3 bucket name but not a unique dynamodb table name so it doesn't make sense to force the dynamodb table to use the same name as the s3 bucket if that's undesired

## references
* closes #87

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -178,7 +180,7 @@ module "dynamodb_table_label" {
 
 resource "aws_dynamodb_table" "with_server_side_encryption" {
   count          = local.dynamodb_enabled && var.enable_server_side_encryption ? 1 : 0
-  name           = module.dynamodb_table_label.id
+  name           = local.dynamodb_table_name
```
[the remark this answers is not in the document]
Comment (aknysh) [reply main.tf:183]: update label module version to the latest 

```
module "dynamodb_table_label" {
  source     = "cloudposse/label/null"
  version    = "0.24.1"
  attributes = compact(concat(var.attributes, ["lock"]))
  context    = module.this.context
  enabled    = local.dynamodb_enabled
}
```

File: variables.tf [hunk 2]
```diff
@@ -164,3 +164,9 @@ variable "dynamodb_enabled" {
   default     = true
   description = "Whether to create the dynamodb table."
 }
+
+variable "dynamodb_table_name" {
+  type        = string
+  default     = null
+  description = "Override the name of the dynamodb table which defaults to using `module.dynamodb_table_label.id`."
```
Comment (aknysh) [inline variables.tf:171]:
```suggestion
  description = "Override the name of the DynamoDB table which defaults to using `module.dynamodb_table_label.id`"
```

**Review by aknysh — COMMENTED** [review]
please see comments

File: main.tf (hunk 1 above)
[the remark this answers is not in the document]
Comment (nitrocode) [reply main.tf:183]: Could we do it in this separate pr?

https://github.com/cloudposse/terraform-aws-tfstate-backend/pull/93

File: variables.tf [hunk 3]
```diff
@@ -164,3 +164,9 @@ variable "dynamodb_enabled" {
   default     = true
   description = "Whether to create the dynamodb table."
 }
+
+variable "dynamodb_table_name" {
+  type        = string
+  default     = null
+  description = "Override the name of the DynamoDB table which defaults to using `module.dynamodb_table_label.id`"
```
Comment (korenyoni) [inline variables.tf:171]:
```suggestion
  description = "Override the name of the DynamoDB table (which defaults to using `module.dynamodb_table_label.id`)."
```

**Review by korenyoni — APPROVED** [review]
LGTM.

**Pull request merged**